ed25519/signer_test.go

```diff
@@ -0,0 +1,295 @@
+				keyId:       testKeyId,
+				keyType:     wrapping.KeyType_ED25519,
+			},
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			assert, require := assert.New(t), require.New(t)
+			gotSigner, err := NewSigner(testCtx, tc.opt...)
+			if tc.wantErr {
+				require.Error(err)
+				assert.Empty(gotSigner)
+				if tc.wantErrIs != nil {
+					assert.ErrorIs(err, tc.wantErrIs)
+				}
+				if tc.wantErrContains != "" {
+					assert.Contains(err.Error(), tc.wantErrContains)
+				}
+				return
+			}
+			require.NoError(err)
+			assert.Equal(tc.wantSigner, gotSigner)
+		})
+	}
+}
+
+func Test_SignVerify(t *testing.T) {
+	t.Parallel()
+	const (
+		testKeyId      = "key-id"
+		testKeyPurpose = wrapping.KeyPurpose_Sign
+	)
+	testPt := []byte("test-plaintext")
+	testCtx := context.Background()
+	testPubKey, testPrivKey, err := ed25519.GenerateKey(rand.Reader)
+	require.NoError(t, err)
+	testSigner, err := NewSigner(testCtx, WithPrivKey(testPrivKey), wrapping.WithKeyId(testKeyId), wrapping.WithKeyPurposes(testKeyPurpose), wrapping.WithKeyType(wrapping.KeyType_ED25519))
+	require.NoError(t, err)
+	testVerifier, err := NewVerifier(testCtx, WithPubKey(testPubKey))
+	require.NoError(t, err)
+	tests := []struct {
+		name            string
+		signer          wrapping.SigInfoSigner
+		msg             []byte
+		sig             *wrapping.SigInfo
+		opt             []wrapping.Option
+		verifier        wrapping.SigInfoVerifier
+		wantErr         bool
+		wantErrIs       error
+		wantErrContains string
+	}{
+		{
+			name:   "success-all-opts",
+			signer: testSigner,
+			msg:    testPt,
+			sig: func() *wrapping.SigInfo {
+				si := TestSigInfo(t, testPrivKey, testPt, wrapping.WithKeyId(testKeyId), wrapping.WithKeyPurposes(wrapping.KeyPurpose_Sign), wrapping.WithKeyType(wrapping.KeyType_ED25519))
+				return si
+			}(),
+			verifier: testVerifier,
+		},
+		{
+			name:            "missing-msg",
+			signer:          testSigner,
+			wantErr:         true,
+			wantErrIs:       wrapping.ErrInvalidParameter,
+			wantErrContains: "missing message",
+		},
+		{
+			name:            "nil-private-key",
+			signer:          &Signer{},
+			msg:             testPt,
+			wantErr:         true,
+			wantErrIs:       wrapping.ErrInvalidParameter,
+			wantErrContains: "missing private key",
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			assert, require := assert.New(t), require.New(t)
+			gotSig, err := tc.signer.Sign(testCtx, tc.msg, tc.opt...)
+			if tc.wantErr {
+				require.Error(err)
+				assert.Empty(gotSig)
+				if tc.wantErrIs != nil {
+					assert.ErrorIs(err, tc.wantErrIs)
+				}
+				if tc.wantErrContains != "" {
+					assert.Contains(err.Error(), tc.wantErrContains)
+				}
+				return
+			}
+			require.NoError(err)
+			assert.Equal(tc.sig, gotSig)
+			gotVerify, err := tc.verifier.Verify(testCtx, tc.msg, gotSig)
+			require.NoError(err)
+			assert.True(gotVerify)
+		})
```

Comment:
This test structure uses the verifier as the oracle for the signer and vice versa - do you think we could add an external oracle that verifies the behavior too? E.g. just a call to the `ed25519.Verofy`. I know that's what the verifier does under the hood but it'd still be nice to have it here as an external trusted test oracle.

Reply:
done.